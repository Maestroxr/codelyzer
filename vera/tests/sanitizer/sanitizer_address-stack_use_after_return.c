int* ptr = 0;
void functionThatEscapesLocalObject() 
{
	int local[ 100 ] = { 0 };
	ptr = &local[ 0 ];
}
int main(int argc, char **argv) 
{
	functionThatEscapesLocalObject();
	return ptr[ argc ];
}