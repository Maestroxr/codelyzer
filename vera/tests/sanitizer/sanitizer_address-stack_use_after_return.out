C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-stack_use_after_return.c:10: Memory:[C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-stack_use_after_return] Stack memory use after function return, suspect variable: 'ptr', sanitizer_address-stack_use_after_return.c:10
	Was accessed by read of size 4 at:
		sanitizer_address-stack_use_after_return.c:10 in function 'main'
	Accessed address is located in stack of thread T0 at offset 36 in frame:
		sanitizer_address-stack_use_after_return.c:3 in function 'functionThatEscapesLocalObject'
:MemoryEnd:
